### PR TITLE
refact(reddit): Clean up subreddit name processing

### PR DESCRIPTION
**Pull request by Wolveric**
Users prefixing their input with r/ when adding Reddit feeds,
has been a persistent issue, enough to the point it had a dedicated note
on the config. page in the control panel.

This PR initially intended to address exclusively this issue,
by adding behaviour to strip leading instances of /r/
from inputs provided when adding subreddit feeds,
but also discovered the functions handling modifying
and removing these feeds, seemed to be performing redundant work
processing the subreddit names, when this work was already performed
when creating the feed.

Consequently, this PR also removes these redundant function calls,
cleaning up the handling of subreddit names in general.

### CODE REVIEW

File: reddit/plugin_web.go [hunk 1]
```diff
@@ -136,10 +136,18 @@ func HandleNew(w http.ResponseWriter, r *http.Request) interface{} {
 		return templateData.AddAlerts(web.ErrorAlert(fmt.Sprintf("Max %d feeds allowed (or %d for premium servers)", GuildMaxFeedsNormal, GuildMaxFeedsPremium)))
 	}
 
+	subreddit := strings.TrimSpace(newElem.Subreddit)
+	subreddit = strings.ToLower(subreddit)
+	subreddit = strings.TrimPrefix(subreddit, "/")
+
+	for ; strings.HasPrefix(subreddit, "r/"); {
```
Comment (SoggySaussages) [inline reddit/plugin_web.go:143]: How about splitting at `/` and only taking the last string?

Comment (jo3-l) [reply reddit/plugin_web.go:143]: To be honest, I think just `subreddit = strings.TrimPrefix(subreddit, "r/")` is enough--no loop. I don't see a real reason to support `r/r/foo`, which is clearly a typo, whereas accepting `r/foo` is a clear improvement.

Comment (ashishjh-bst) [reply reddit/plugin_web.go:143]: > To be honest, I think just `subreddit = strings.TrimPrefix(subreddit, "r/")` is enough--no loop. I don't see a real reason to support `r/r/foo`, which is clearly a typo, whereas accepting `r/foo` is a clear improvement.

I agree with this.

Comment (Wolveric) [reply reddit/plugin_web.go:143]: Yeah, I didn't love the idea either, so I'll drop the `for` loop from the behaviour.

**Pull request merged**